Add an option to fail the test if it times out and enable on Travis

ylemkimon opened the pull request:
* Added `fail_on_timeout` option to fail the test by raising TimeOutError and print traceback if it times out.
* Enabled 595 seconds timeout on Travis builds. ([Example](https://travis-ci.org/ylemkimon/sympy/jobs/280894080))

See comments of #13355.

ylemkimon committed:
Remove unused functions from test_failing_integrals

Removed unused run_with_timeout and related from test_failing_integrals.

ylemkimon committed:
Add an option to fail the test if it times out

Added an fail_on_timeout option to fail the test by raising TimeOutError
and print traceback if it times out.

ylemkimon committed:
Enable timeout on Travis

Enabled 595 seconds timeout on Travis builds and made fail the test and
print traceback if it times out.

smichr commented on sympy/utilities/runtests.py:
```diff
@@ -1059,7 +1067,8 @@ def __init__(self, reporter, kw="", post_mortem=False,
 
-    def test(self, sort=False, timeout=False, slow=False, enhance_asserts=False):
+    def test(self, sort=False, timeout=False, fail_on_timeout=False, slow=False,
+            enhance_asserts=False):
```
To preserve arguments passed positionally, this new parameter should be added to the end of the parameter list, shouldn't it?

smichr commented on sympy/utilities/runtests.py:
```diff
@@ -1113,7 +1123,8 @@ def visit_Assert(self, stmt):
         return fix_missing_locations(new_tree)
 
-    def test_file(self, filename, sort=True, timeout=False, slow=False, enhance_asserts=False):
+    def test_file(self, filename, sort=True, timeout=False, fail_on_timeout=False,
```
ditto regarding parameter order

ylemkimon replied:
@smichr Fixed.

ylemkimon committed:
Change argument positions

Changed fail_on_timeout argument positions to the end to preserve
arguments passed positionally.

smichr commented:
Unless the test that is timing out passes we won't be able to commit this.
